miden-lib/asm/miden/kernels/tx/prologue.masm

```diff
@@ -364,16 +365,23 @@ proc.process_acct_data
     padw padw padw
     # => [ZERO, ZERO, ZERO, acct_data_ptr]
 
-    # read the account data
+    # load, save, and hash the data
     adv_pipe hperm adv_pipe hperm
-    # => [ACT_DATA', ACCT_HASH, CAP', act_data_ptr']
+    # => [RATE1, RATE0, CAPACITY, act_data_ptr']
 
-    # extract digest from hasher rate elements (h_0, ..., h_3) and drop pointer
+    # extract digest from hasher rate state and drop pointer
     dropw swapw dropw movup.4 drop
     # => [ACCT_HASH]
 
-    # check if the account is new
-    padw exec.memory::get_init_acct_hash eqw movdn.8 dropw dropw
+    # Validate the account
+    # ---------------------------------------------------------------------------------------------
+
+    # It is a new account if the global input INITIAL_ACCOUNT_HASH was set to ZERO.
+    padw exec.memory::get_init_acct_hash eqw
+    # => [is_new, INITIAL_ACCOUNT_HASH, ZERO, ACCT_HASH]
+
+    # clean the stack
+    movdn.8 dropw dropw
     # => [is_new, ACCT_HASH]
 
     # ingest storage slot type data
```

Comment:
I tried moving the ingest after this if to group the state update calls, but the tests are failing. I didn't look deeper into it. Just moved the call back to its original place.

Reply:
I think this is because `validate_new_account` checks that storage layout is valid. Maybe worth adding a comment about this.